packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -143,16 +143,19 @@ const App = (props) => {
         l10nConfig: site.l10n
     })
 
-    // Workaround for infinite loop if default site locale translation file is missing
-    const isDefaultTranslation = new RegExp(
-        `/static/translations/compiled/${site.l10n.defaultLocale}\\.json$`
-    ).test(location?.pathname)
+    // If the translation file exists, it'll be served directly from static folder (and won't reach this code here).
+    // However, if the file is missing, the App would render a 404 page.
+    const is404ForMissingTranslationFile = /\/static\/translations\/compiled\/[^.]+\.json$/.test(
```

Comment:
Will these paths work as expected on Windows?

Reply:
@adamraya I think it should work on Windows too. Because unlike file system requests, this fetch request has its own url that is consistent across windows, Mac and others.

Reply:
I think so. This is a URL path, not a file path

Reply:
yep, you're right I missed we're working here with a fetch request instead of file systems paths.